app/helpers/authentication_helper.rb

```diff
@@ -33,23 +37,38 @@ def waiting_on_first_user?
     SiteConfig.waiting_on_first_user
   end
 
-  def disable_email_tooltip_class
+  def email_auth_disable_btn_tooltip_class
     SiteConfig.invite_only_mode || authentication_enabled_providers.none? ? "crayons-tooltip" : ""
   end
 
-  def disable_email_tooltip_content
-    SiteConfig.invite_only_mode || authentication_enabled_providers.none? ? disable_email_auth_tooltip_text : ""
+  def auth_provider_enable_btn_tooltip_class
+    SiteConfig.invite_only_mode ? "crayons-tooltip" : ""
+  end
+
+  def auth_provider_disable_btn_tooltip_class
+    !SiteConfig.allow_email_password_login && authentication_enabled_providers.count == 1 ? "crayons-tooltip" : ""
   end
 
-  def disable_button_class
+  def email_auth_disable_btn_disable_class
     SiteConfig.invite_only_mode || authentication_enabled_providers.none? ? "disabled" : ""
   end
 
-  def disable_email_auth_tooltip_text
+  def auth_provider_enable_btn_disable_class
+    SiteConfig.invite_only_mode ? "disabled" : ""
+  end
+
+  def auth_provider_disable_btn_disable_class
+    !SiteConfig.allow_email_password_login && authentication_enabled_providers.count == 1 ? "disabled" : ""
```

Comment:
**Suggestion (nitpick)** This is the logic check as line 49 and line 68, consider extracting it to a method

Reply:
Done!